### PR TITLE
Update tests for caliper delivery point

### PR DESCRIPTION
**Pull request by mumarkhan999**

### CODE REVIEW

File: openedx_caliper_tracking/tests/tests.py [hunk 1]
```diff
@@ -81,11 +81,43 @@ class CaliperDeliveryTestCase(TestCase):
     )
     def test_event_is_not_called_without_settings(self, delivery_mock):
         """
-        If the settings CALIPER_DELIVER_ENDPOINT and CALIPER_DELIVERY_AUTH_TOKEN are
-        not present then the API call shouldn't fire
+        If the settings ENABLE_CALIPER_EVENTS_DELIVERY, CALIPER_DELIVER_ENDPOINT
```
Comment (danialmalik) [inline openedx_caliper_tracking/tests/tests.py]:
```suggestion
        If the settings ENABLE_CALIPER_EVENTS_DELIVERY, CALIPER_DELIVERY_ENDPOINT
```

File: openedx_caliper_tracking/tests/tests.py [hunk 2]
```diff
@@ -81,11 +81,43 @@ class CaliperDeliveryTestCase(TestCase):
...
+        and CALIPER_DELIVERY_AUTH_TOKEN are not present then the API call shouldn't fire
         """
-        del settings.CALIPER_DELIVERY_ENDPOINT
-        del settings.CALIPER_DELIVERY_AUTH_TOKEN
+        if settings.FEATURES.get('ENABLE_CALIPER_EVENTS_DELIVERY'):
```
Comment (danialmalik) [inline openedx_caliper_tracking/tests/tests.py:92]: Please use patch.dict for changing the dict values.
https://docs.python.org/3/library/unittest.mock.html#patch-dict

**Review by danialmalik — CHANGES_REQUESTED** [review]

File: openedx_caliper_tracking/tests/tests.py (hunk 1 above)
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py]: done.

File: openedx_caliper_tracking/tests/tests.py (hunk 2 above)
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:92]: done.

Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:92]: In this scenario, it is not possible but I have used it wherever it is possible.

File: openedx_caliper_tracking/tests/tests.py [hunk 3]
```diff
@@ -81,11 +81,46 @@ class CaliperDeliveryTestCase(TestCase):
     )
     def test_event_is_not_called_without_settings(self, delivery_mock):
         """
-        If the settings CALIPER_DELIVER_ENDPOINT and CALIPER_DELIVERY_AUTH_TOKEN are
-        not present then the API call shouldn't fire
+        If the settings ENABLE_CALIPER_EVENTS_DELIVERY, CALIPER_DELIVERY_ENDPOINT
+        and CALIPER_DELIVERY_AUTH_TOKEN are not present then the API call shouldn't fire.
+        """
+        if settings.FEATURES.get('ENABLE_CALIPER_EVENTS_DELIVERY'):
+            settings.FEATURES.pop('ENABLE_CALIPER_EVENTS_DELIVERY')
```
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py:93]: why this? you should override settings.

**Review by imhassantariq — APPROVED** [review]

**Review by tehreem-sadat — APPROVED** [review]

File: openedx_caliper_tracking/tests/tests.py [hunk 4]
```diff
@@ -81,11 +81,46 @@ class CaliperDeliveryTestCase(TestCase):
...
+    )
+    @mock.patch.dict(
+        'django.conf.settings.FEATURES',
+        {'ENABLE_CALIPER_EVENTS_DELIVERY': False},
+        clear=False
+    )
+    def test_event_is_not_called_with_unset_delivery_flag(self, delivery_mock):
+        """
+        If the feature flag ENABLE_CALIPER_EVENTS_DELIVERY not set means False
+        then the API call shouldn't fire.
```
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py]: can we write simple docstring? 
```
Test that API won't call when `ENABLE_CALIPER_EVENTS_DELIVERY` is not set.
```

File: openedx_caliper_tracking/tests/tests.py (hunk 3 above)
Comment (tasawernawaz) [reply openedx_caliper_tracking/tests/tests.py:93]: usually, we start docstring in tests with `Test` or `Verify` for consistency and readability.

File: openedx_caliper_tracking/tests/tests.py [hunk 5]
```diff
@@ -102,7 +137,12 @@ def test_event_is_not_called_without_settings(self, delivery_mock):
     )
     @override_settings(
         CALIPER_DELIVERY_ENDPOINT='http://localhost:3000',
-        CALIPER_DELIVERY_AUTH_TOKEN='test_auth_token',
+        CALIPER_DELIVERY_AUTH_TOKEN='test_auth_token'
```
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py:144]: why change on this line ?

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: openedx_caliper_tracking/tests/tests.py (hunk 5 above)
Comment (danialmalik) [reply openedx_caliper_tracking/tests/tests.py:144]: trailing `,` is removed

File: openedx_caliper_tracking/tests/tests.py (hunk 3 above)
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:93]: > why this? you should override settings.

Actually here I am deleting these settings if exists. Before these changes, our tests were failing as we were trying to delete the settings which weren't present in the settings object.

File: openedx_caliper_tracking/tests/tests.py (hunk 5 above)
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:144]: just for the consistency, I removed trailing commas

**Comment by mumarkhan999** [comment]
@tasawernawaz All the docstrings have been updated in the latest commit.

**Review by danialmalik — APPROVED** [review]

**Comment by mumarkhan999** [comment]
@tasawernawaz waiting for your approval/review.

**Review by tasawernawaz — APPROVED** [review]

**Pull request merged**